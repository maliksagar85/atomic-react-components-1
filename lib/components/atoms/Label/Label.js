import React from 'react';
import PropTypes from 'prop-types';

import { getLabelFromChildren } from '../../../core/utils/elementLocator';

const Label = ({ className, htmlFor, children, elementLocator, ...others }) => (
  /* eslint-disable jsx-a11y/label-has-for */
  // FIXME: Remove the eslint comment above
  <label
    {...others}
    className={className}
    htmlFor={htmlFor}
    data-locator={elementLocator || `label-${htmlFor}-${getLabelFromChildren(children)}`}
  >
    {children}
  </label>
);

Label.propTypes = {
  className: PropTypes.string,
  elementLocator: PropTypes.string,
  htmlFor: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

Label.defaultProps = {
  className: '',
  elementLocator: '',
};

export default Label;
