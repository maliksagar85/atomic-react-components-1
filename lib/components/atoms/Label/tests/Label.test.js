import React from 'react';
import { shallow, render } from 'enzyme';

import Label from '../Label';

describe('Label Component', () => {
  let label = '';

  test('should render correctly', () => {
    label = shallow(<Label htmlFor="userid"> User Name </Label>);
    expect(label).toMatchSnapshot();
  });

  test('should check the data attribute of label', () => {
    const sampleProps = { htmlFor: 'userid', elementLocator: 'my-custom-locator' };
    const image = render(<Label {...sampleProps} />);

    expect(image.attr('data-locator')).toEqual(sampleProps.elementLocator);
  });
});
