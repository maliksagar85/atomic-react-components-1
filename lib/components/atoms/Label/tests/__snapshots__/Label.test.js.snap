// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render correctly 1`] = `
<label
  className=""
  data-locator="label-userid--user"
  htmlFor="userid"
>
   User Name 
</label>
`;
